From 2b7d1d2720631a45743ab898ca169edd178136d8 Mon Sep 17 00:00:00 2001
From: Katarina Behrens <Katarina.Behrens@cib.de>
Date: Fri, 8 Jun 2018 15:04:13 +0200
Subject: [PATCH] Add overloaded read/send funcs for uint64_t (conditionally)

otherwise build fails on 32bit platforms as multiple matching
overloaded func are found. Thanks _rene_ and mst_ for helping
me to figure this out

Change-Id: Ief4571ad735ad4efea9ddc70daea45885e5954c7
---
 vcl/unx/gtk3_kde5/filepicker_ipc_commands.hxx | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/vcl/unx/gtk3_kde5/filepicker_ipc_commands.hxx b/vcl/unx/gtk3_kde5/filepicker_ipc_commands.hxx
index 8ce8cfa9e78d..1acef8caf48e 100644
--- a/vcl/unx/gtk3_kde5/filepicker_ipc_commands.hxx
+++ b/vcl/unx/gtk3_kde5/filepicker_ipc_commands.hxx
@@ -100,6 +100,14 @@ inline void readIpcArg(std::istream& stream, sal_uIntPtr& value)
     stream.ignore(); // skip space
 }
 
+#if SAL_TYPES_SIZEOFPOINTER == 4
+inline void readIpcArg(std::istream& stream, uint64_t& value)
+{
+    stream >> value;
+    stream.ignore(); // skip space
+}
+#endif
+
 inline void readIpcArgs(std::istream& /*stream*/)
 {
     // end of arguments, nothing to do
@@ -135,6 +143,10 @@ inline void sendIpcArg(std::ostream& stream, sal_Int16 value) { stream << value
 
 inline void sendIpcArg(std::ostream& stream, sal_uIntPtr value) { stream << value << ' '; }
 
+#if SAL_TYPES_SIZEOFPOINTER == 4
+inline void sendIpcArg(std::ostream& stream, uint64_t value) { stream << value << ' '; }
+#endif
+
 inline void sendIpcArgsImpl(std::ostream& stream)
 {
     // end of arguments, flush stream
-- 
2.17.1

